tensorboard/webapp/persistent_settings/_redux/persistent_settings_effects_test.ts

```diff
@@ -0,0 +1,211 @@
+
+    await TestBed.configureTestingModule({
+      imports: [
+        PersistentSettingsTestingDataSourceModule,
+        PersistentSettingsConfigModule.defineGlobalSetting(
+          setSmoothingSelector
+        ),
+        PersistentSettingsConfigModule.defineGlobalSetting(setIgnoreOutliers),
+      ],
+      providers: [
+        provideMockActions(action),
+        provideMockStore(),
+        PersistentSettingsEffects,
+        PersistentSettingsConfigModule,
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<any>>(Store) as MockStore<any>;
+    dispatchSpy = spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      actualActions.push(action);
+    });
+    effects = TestBed.inject(PersistentSettingsEffects);
+    const dataSource = TestBed.inject(PersistentSettingsTestingDataSource);
+    getSettingsSpy = spyOn(dataSource, 'getSettings').and.returnValue(of({}));
+    setSettingsSpy = spyOn(dataSource, 'setSettings').and.returnValue(EMPTY);
+  });
+
+  afterEach(fakeAsync(() => {
+    discardPeriodicTasks();
+  }));
+
+  describe('#initializeAndUpdateSettings$', () => {
+    beforeEach(() => {
+      effects.initializeAndUpdateSettings$.subscribe();
+    });
+
+    describe('on init', () => {
+      it('fetches user settings and emits an aciton', () => {
+        getSettingsSpy.and.returnValue(
+          of({
+            ignoreOutliers: false,
+          })
+        );
+        action.next(TEST_ONLY.initAction());
+
+        expect(actualActions).toEqual([
+          globalSettingsLoaded({
+            partialSettings: {
+              ignoreOutliers: false,
+            },
+          }),
+        ]);
+      });
+
+      it('subscribes to selector changes after initial setting is read', fakeAsync(() => {
+        store.overrideSelector(setSmoothingSelector, {scalarSmoothing: 0.1});
+        const getSettingsSubject = new ReplaySubject(1);
+
+        getSettingsSpy.and.returnValue(getSettingsSubject);
+        action.next(TEST_ONLY.initAction());
+
+        tick();
+        store.overrideSelector(setSmoothingSelector, {scalarSmoothing: 0.3});
+        store.refreshState();
+        tick(TEST_ONLY.DEBOUNCE_PERIOD_IN_MS * 2);
+        expect(setSettingsSpy).not.toHaveBeenCalled();
+
+        getSettingsSubject.next({ignoreOutliers: false});
+        getSettingsSubject.complete();
+        tick();
+        store.overrideSelector(setSmoothingSelector, {scalarSmoothing: 0.5});
+        store.refreshState();
+        tick(TEST_ONLY.DEBOUNCE_PERIOD_IN_MS * 2);
+        expect(setSettingsSpy).toHaveBeenCalledOnceWith({
+          scalarSmoothing: 0.5,
+        });
+      }));
+
+      it('ignores value emitted from initial subscription', fakeAsync(() => {
+        store.overrideSelector(setSmoothingSelector, {scalarSmoothing: 0.1});
+
+        getSettingsSpy.and.returnValue(of({}));
+        action.next(TEST_ONLY.initAction());
+
+        tick();
+        tick(TEST_ONLY.DEBOUNCE_PERIOD_IN_MS * 2);
```

Comment:
why we need two `tick()` here? omit the first one?

Reply:
It was unnecessary. Removed. Thanks for catching that.